deploy/docker/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Docker image for the primary terria map application server
-FROM node:16
+FROM node:16-slim
 
 RUN mkdir -p /usr/src/app && mkdir -p /etc/config/client
 WORKDIR /usr/src/app/component
```

Comment:
I'm not against the change per se, but there are a lot of things I find confusing.

Is it the terriajs specified in package.json that gets installed, or does this pull from the TerriaJS repository? What creates the `node_modules`  that are copied into /usr/src/app? Is there documentation for how to rebuild the release image locally without using Github actions?

Why does this Dockerfile use `/usr/src/app` and the Dockerfile in the root of this repository use `/app`?

Reply:
I've added an ADR (architecture decision record) explaining why I'm proposing this. Please comment on that, or here if you are still confused or disagree.

This dockerfile copies from a "context" that is created by `create-docker-context.js` - which is a script that copies only production dependencies and build artifacts to make a docker image. It's non-standard, but gets around running dependency installation and compilation on emulated architectures. The terriajs that is used is whichever is installed at the time that `yarn docker-build-prod` is run. We usually run releases from GitHub Actions, or if we need to build docker images locally we use a clean clone of TerriaMap. Both methods will always be a clean install pulling the terriajs version specified in package.json.

I think it's more normal to use /usr/src/app than /app

Reply:
Based on the description, this seems like a great idea.

Are the steps for how to build the docker image locally written down somewhere?

Reply:
Ok, ADR actually in this PR now. Sorry if you tried looking before, I forgot to push it.

As for building the docker image locally, this would be a separate command `yarn docker-build-local`, which doesn't do multi-arch builds. I haven't tested that one yet.

Reply:
I rebuild the release image to get security fixes and a few other local modifications, so I prefer a local build process that is as close to the release-image as possible so I don't have to re-do all the general quality assurance you have already done for the release.

I don't use multi-arch builds personally so not sure how important that is in the grand scheme of things; if I suddenly get ARM machines I can do separate builds on ARM/X86 and then stitch together the two images with crane or some other tool.